### DISCUSSION TITLE
Update password through /self-service/settings is very slow

### DISCUSSION THREAD

**Discussion by ijalgos** [ory/kratos#2789] [Q&A]
Hi everyone and thank you for this amazing project. We are running kratos in a particular isolated environment with a slow network. Updating password through/self-service/settings cause huge delays in 20s to 60s. The assumption was that the problem was in the HaveIBeenPwned API but setting `export SELFSERVICE_METHODS_PASSWORD_CONFIG_HAVEIBEENPWNED_ENABLED=false` doesn't help.  

May I ask what can cause this kind of delay?

### Tech details

```
OS: Linux
Runtime: Docker
Version: oryd/kratos:v0.10.1
```

### Request

```
curl 'https://internal.site/api/.ory/self-service/settings?flow=09fbf919-9d87-4c2c-a21f-c1896a829c8c' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9,ru;q=0.8' \
  -H 'Cache-Control: no-cache' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json' \
  -H 'Cookie: <Reduced>' \
  -H 'Origin: https://internal.site' \
  -H 'Pragma: no-cache' \
  -H 'Referer: https://internal.site/settings/security' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-origin' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/105.0.0.0 Safari/537.36' \
  -H 'sec-ch-ua: "Google Chrome";v="105", "Not)A;Brand";v="8", "Chromium";v="105"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"csrf_token":"F10Q5pLeIxBdpmpWZAtxX1h5LXBjr7TM0Ys8XBVYV6ffdUTZ8Vdn7CCbW1w4kmnPZaT+wYfTNk1SipHHx6zxgg==","password":"supersecret","method":"password"}' \
  --compressed
```

**Comment by vinckr** [comment]
Hey ijalgos, 
this is hard to answer without knowing more about your network & setup.
Are the other flows completing without huge delays?  
Do you see anything in the logs that is out of the ordinary when you make that request?  
What kind of frontend are you running?  
Btw, if you want to offload these things (latency, setup, etc.) on us, you can also check out [Ory Cloud](https://www.ory.sh/cloud/).
